volatility3/framework/plugins/linux/lsblk.py

```diff
@@ -0,0 +1,388 @@
+        return dev & 0xFF
+
+    def _get_read_only(self, vmlinux, gendisk):
+        # before v5.11 has a different check for read only that uses hd_struct
+        if not vmlinux.has_type("hd_struct"):
+            GD_READ_ONLY = 1
+            read_only = (
+                gendisk.part0.bd_read_only
+                or (gendisk.state & (1 << GD_READ_ONLY)) != 0
+            )
+        else:
+            read_only = bool(gendisk.part0.policy)
+        return read_only
+
+    def _get_block_device(self, vmlinux, devt):
+        """In linux versions before 5.11, this function is how you get the block_device struct. Mimics the
+        function struct block_device* bdget(dev_t dev) https://elixir.bootlin.com/linux/v5.4/source/fs/block_dev.c#L900
+        """
+
+        blockdev_superblock = vmlinux.object_from_symbol("blockdev_superblock")
+        inode_hashtable = vmlinux.object_from_symbol("inode_hashtable")
+
+        # This the hash functions used to get the inode number https://elixir.bootlin.com/linux/v5.4/source/fs/block_dev.c#L867
+        val = self._major(devt) + self._minor(devt)
+
+        # Need these next 4 values for this hash function: https://elixir.bootlin.com/linux/v5.4/source/fs/inode.c#L474
+        i_hash_mask = vmlinux.object_from_symbol("i_hash_mask")
+        i_hash_shift = vmlinux.object_from_symbol("i_hash_shift")
+
+        # in linux 4.7, the GOLDEN_RATIO_PRIME constant changes. This struct gains a member from 4.6 to 4.7
+        if vmlinux.get_type("file_operations").has_member("iterate_shared"):
+            GOLDEN_RATIO_PRIME = GOLDEN_RATIO_PRIME_AFTER_4_7
+        else:
+            GOLDEN_RATIO_PRIME = GOLDEN_RATIO_PRIME_BEFORE_4_7
+
+        L1_CACHE_BYTES = 1 << 6
+
+        # Previously mentioned hash function
+        val = (val * blockdev_superblock) ^ (GOLDEN_RATIO_PRIME + val) // L1_CACHE_BYTES
+        val = val ^ ((val ^ GOLDEN_RATIO_PRIME) >> i_hash_shift)
+        val = val & i_hash_mask
+
+        bucket_size = vmlinux.get_type("hlist_head").size
+        hlist_head_offset = inode_hashtable + (val * bucket_size)
+
+        hlist_head = vmlinux.object(
+            object_type="hlist_head",
+            offset=hlist_head_offset,
+            absolute=True,
+        )
+        hlist_node = hlist_head.first
+        if not hlist_node:
+            return 0
+        while hlist_node:
+            inode = linux.LinuxUtilities.container_of(hlist_node, "inode", "i_hash", vmlinux)
+            # These checks are from the find_inode function https://elixir.bootlin.com/linux/v5.4/source/fs/inode.c#L805
+            # Trying to find the inode with the correct device number and calculated inode number from the inode cache
+            if inode.i_sb == blockdev_superblock and inode.i_rdev == devt:
+                break
+            hlist_node = hlist_node.next
+
+        return inode.i_bdev
+
+    def _device_iterator(self, context, klist_devices):
+        vmlinux = context.modules[self.config["kernel"]]
```

Comment:
Please could we parameterize this on `kernel_module_name` rather than going to the config?  It'll make the function more usable outside of the bounds of just this plugin.